### PR TITLE
unXFAIL two tests in test_wester

### LINKED ISSUE #7157: correct integrate() result in python2.7.4, but wrong in python3.3
```
Wester test_V12() in test_wester.py

Tested in Ubuntu 13.04.

Result is correct in python2.7.4:

Python 2.7.4 (default, Sep 26 2013, 03:20:56) 
[GCC 4.7.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import *
>>> x = symbols('x')
>>> integrate(1/(5 + 3*cos(x) + 4*sin(x)), x)
-1/(tan(x/2) + 2)

But incorrect on python3.3.1:

Python 3.3.1 (default, Sep 25 2013, 19:30:50) 
[GCC 4.7.3] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import *
>>> x = symbols('x')
>>> integrate(1/(5 + 3*cos(x) + 4*sin(x)), x)
tan(x/2)/(2*tan(x/2) + 4)
```

Original issue for #7157: http://code.google.com/p/sympy/issues/detail?id=4058
Original author: https://code.google.com/u/118279007905392951376/

### PR DESCRIPTION
**Pull request by debugger22**
Closes https://github.com/sympy/sympy/issues/7157

### CODE REVIEW

File: sympy/core/tests/test_wester.py [hunk 1]
```diff
@@ -2341,11 +2339,8 @@ def test_V11():
             log(((tan(x/2) + 1)/(tan(x/2) + 7))**Rational(1, 3)))
 
 
-@XFAIL
 def test_V12():
     r1 = integrate(1/(5 + 3*cos(x) + 4*sin(x)), x)
-    # Correct result in python2.7.4 wrong result in python3.3.1
-    # https://github.com/sympy/sympy/issues/7157
```
Comment (toolforger) [inline sympy/core/tests/test_wester.py]: How about leaving in a reference to #7157 so if this regresses, people know where to look for more information?

Comment (debugger22) [inline sympy/core/tests/test_wester.py]: Done.

**Comment by debugger22** [comment]
@toolforger Are you +1 now?

**Comment by toolforger** [comment]
I'd have left in the reference to the issue report.
Aaron was suggesting it's related to randomness, not to Python versions; that aspect seems to be unresolved even here, so it's important to keep that hint accessible so people know what to check if this should regress.

**Comment by debugger22** [comment]
@skirpichev Was https://github.com/sympy/sympy/commit/fad74a85495ca8b60fe74116fe91af22f8417c7a reverted?

**Comment by asmeurer** [comment]
The tests are run with random seeds so if it is still an issue it should show up.

**Comment by debugger22** [comment]
It passed once. Should I restart it?

**Comment by skirpichev** [comment]
> @skirpichev Was fad74a8 reverted?

I don't see this commit in the history.

**Comment by debugger22** [comment]
Me too but then why it has `sympy/sympy` in the URL of the commit.

**Comment by skirpichev** [comment]
No idea.  Perhaps, this is commit in my fork (now deleted).

**Comment by toolforger** [comment]
From the old messages, it seemed to me that the randomness has just selects one or the other of two equally valid results.
Without detail knowledge of what exactly is happening there, we don't know the probabilities. With a 50:50 probability, you'd have a confidence of 1:1023 that the fix works if the tests pass; if the probability is 99.999:0.001, you'd have to run many more tests to be reasonably confident, and you wouldn't know when to stop testing.
IOW more testing can only reveal a problem but it can't make you confident.

Since we haven't seen wrong results, we probably can still just run the test as it is and revisit the issue if we ever see a problem. A few more test runs (locally, I guess) should give you enough confidence that it won't happen too soon on Travis, or for anybody who runs the tests locally.
(Still, it would be nice to know what was causing the problem then, and whether it's fixed or not. I don't know how to properly deal with heuristics in testing though - how do you find out whether two results are equivalent if the equivalence algorithm is part of the stuff you're testing?)

**Comment by skirpichev** [comment]
> With a 50:50 probability, you'd have a confidence of 1:1023 that the fix works if the tests pass

Looks as arbitrary numbers for me...  (BTW, how many words and how little sense.)

@debugger22, try something like this (works fine for me):
```
$ ./bin/test --rerun=10 -k test_V12 sympy/core/tests/test_wester.py
```
Maybe some stuff in heurisch still depends on dict/set's sorting.  Maybe not, i.e. after 2e06e1a (I have failure for above command before this commit.).  I bet it was fixed by @smichr ;)

**Comment by debugger22** [comment]
It fails.
```
========================== test process starts ===========================
executable:         /usr/bin/python  (2.7.5-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
random seed:        19341154
hash randomization: on (PYTHONHASHSEED=3289598534)

sympy/core/tests/test_wester.py[1] F                                [FAIL]

__________________________________________________________________________
________________ sympy/core/tests/test_wester.py:test_V12 ________________
  File "/home/mrsud/workspace/sympy/sympy/core/tests/test_wester.py", line 2345, in test_V12
    assert r1 == -1/(tan(x/2) + 2)
AssertionError

```
As @asmeurer already pointed out, it's independent of Python version too.

Perhaps it can be closed.

**Comment by skirpichev** [comment]
Oh, it seems I have found this "source of randomness".

**Comment by toolforger** [comment]
Am 12.04.2015 um 01:06 schrieb Sergey B Kirpichev:
>> With a 50:50 probability, you'd have a confidence of 1:1023 that the fix works if the tests pass
>
> Looks as arbitrary numbers for me...

Yeah. Arbitrary numbers are often a stand-in for free variables. So:
If you have a k:(1-k) probability, the confidence is k**N after N tests.

 > (BTW, how many words and how little sense.)

(BTW, seems insufficient understanding has elicited your troll side again.)

File: sympy/core/tests/test_wester.py [hunk 2]
```diff
@@ -2341,10 +2339,8 @@ def test_V11():
             log(((tan(x/2) + 1)/(tan(x/2) + 7))**Rational(1, 3)))
 
 
-@XFAIL
 def test_V12():
     r1 = integrate(1/(5 + 3*cos(x) + 4*sin(x)), x)
-    # Correct result in python2.7.4 wrong result in python3.3.1
```
Comment (skirpichev) [inline sympy/core/tests/test_wester.py]: BTW, there is no "wrong result".  I get `tan(x/2)/(2*tan(x/2) + 4)`, that's same antiderivative as expr below.

Comment (debugger22) [inline sympy/core/tests/test_wester.py]: Yes. The comment is misleading. It was already pointed out on the issue.

File: sympy/core/tests/test_wester.py [hunk 3]
```diff
@@ -2093,9 +2093,7 @@ def test_T9():
     assert limit(hyper((1, k), (1,), z/k), k, oo) == exp(z)
 
 
-@XFAIL
 def test_T10():
-    # raises PoleError should return euler-mascheroni constant
     limit(zeta(x) - 1/(x - 1), x, 1)
```
Comment (skirpichev) [inline sympy/core/tests/test_wester.py]: This test should be fixed instead, not UnXFAIL'ed.

BTW, here is a pretty-printing bug too:
```python
In [1]: limit(zeta(x) - 1/(x - 1), x, 1)
Out[1]: 
              1  
 lim ζ(x) - ─────
x─→1⁺       x - 1

In [2]: print _
Limit(zeta(x) - 1/(x - 1), x, 1)
```

**Pull request closed without merging**